### ISSUE TITLE
ui: Failed Node deployment jobs may prevent access to Deploy button

### ISSUE THREAD

**Issue by gdemonet** [scality/metalk8s#2602]
**Component**: ui

**What happened**:

After having failed a node deployment using the UI (**Deploy** button), the **Deploying...** state of this button didn't change, making it impossible to retry deployment.
Changing to a private navigation window fixed the issue.

**What was expected**:

Failed jobs should be cleared from the browser state and the **Deploy** button should be made available (or even maybe cancellable?) when a failure happened or the saved job doesn't exist anymore.

**Steps to reproduce**: Not sure. Maybe try manually adding a fake JID (pointing to an actual Node) in your `LocalStorage` and see how that behaves.

**Resolution proposal** (optional):

Probably we should handle "not found" Job IDs when going through the supposedly active Jobs stored in local storage, and mark the lost ones as inactive, at least.
If we find a reproducing strategy, let's make it an automatic test to ensure our code is robust enough.

**Comment by slaperche-scality** [comment]
Remind me of https://github.com/scality/metalk8s/issues/2443, which was a bit similar (but hard to reproduce)

**Comment by gdemonet** [comment]
You're right, and this bug happened yesterday on 2.4.3, where the fix isn't merged yet. Maybe there's nothing to do outside backporting to 2.4.4.

**Comment by thomasdanan** [comment]
maybe also linked to https://github.com/scality/metalk8s/issues/2803
